lib/features/mailbox/presentation/widgets/mailbox_folder_tile_builder.dart

```diff
@@ -336,7 +391,9 @@ class MailBoxFolderTileBuilder {
 
   Widget _buildMailboxIcon() {
     return SvgPicture.asset(
-        _mailboxNode.item.getMailboxIcon(_imagePaths),
+        _mailboxNode.item.isChildOfTeamMailboxes 
+         ? _mailboxNode.item.getChildOfTeamMailboxesIcon(_imagePaths)
+         : _mailboxNode.item.getMailboxIcon(_imagePaths),
```

Comment:
IMO, checking condition and get the icon should be executed inside `getMailboxIcon`